server/adaptors/integrations/__test__/builder.test.ts

```diff
@@ -121,22 +131,24 @@ describe('IntegrationInstanceBuilder', () => {
         dataSource: 'instance-datasource',
         name: 'instance-name',
       };
-      sampleIntegration.deepCheck = jest.fn().mockResolvedValue(false);
+      sampleIntegration.deepCheck = jest
+        .fn()
+        .mockResolvedValue({ ok: false, error: new Error('Mock error') });
```

Comment:
I see the introduction of Result further down in the changes.
Should these uses also be converted to Result?

Reply:
Only have converted part of it to result to avoid making the PR too large, intend to do more later